projects/resettlement-passport-and-delius/src/main/kotlin/uk/gov/justice/digital/hmpps/api/model/CreateContact.kt

```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.digital.hmpps.api.model
+
+import io.swagger.v3.oas.annotations.media.Schema
+import jakarta.validation.constraints.NotBlank
+import java.time.ZonedDateTime
+
+data class CreateContact(
+    @Schema(
+        type = "string",
+        example = "RP9",
+        description = "Nomis contact type. Must either RP9 or RP10"
+    )
+    val type: Type,
+    val dateTime: ZonedDateTime,
+    @field:NotBlank
+    val notes: String,
+) {
+    enum class Type {
+        RP9,
+        RP10,
+    }
```

Comment:
Sorry just thought of this - it seems like RP9 & RP10 are internal Delius codes, maybe we should map them from a friendlier name instead so that Resettlement Passport doesn't need to map it themselves?

For example,
```suggestion
    enum class CaseNoteType(contactTypeCode: String) {
        IMMEDIATE_NEEDS_REPORT("RP9"),
        PRE_RELEASE_REPORT("RP10"),
    }
```

Reply:
yeah i did think of this actually. Looks like more changes may be required anyway

Reply:
Made these changes along with others